File: onebot/src/main/java/cn/evole/onebot/sdk/event/notice/group/GroupAdminNoticeEvent.java
package cn.evole.onebot.sdk.event.notice.group;

import cn.evole.onebot.sdk.event.notice.NoticeEvent;
import com.google.gson.annotations.SerializedName;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

/**
 * Created on 2022/7/8.
 *
 * @author cnlimiter
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder(toBuilder = true)
@EqualsAndHashCode(callSuper = true)
public class GroupAdminNoticeEvent extends NoticeEvent {

    /**
     * set、unset
     * 事件子类型, 分别表示设置和取消管理
     */
    @SerializedName( "sub_type")
    public String subType;

    /**
     * 群号
     */
    @SerializedName( "group_id")
    public long groupId;

    /**
     * 管理员 QQ 号
     */


}

File: onebot/src/main/kotlin/sdk/event/notice/group/GroupAdminNoticeEvent.kt
package cn.evolvefield.onebot.sdk.event.notice.group

import cn.evolvefield.onebot.sdk.event.notice.NoticeEvent
import com.google.gson.annotations.SerializedName
import lombok.AllArgsConstructor
import lombok.Data
import lombok.EqualsAndHashCode
import lombok.NoArgsConstructor

@Data
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
class GroupAdminNoticeEvent : NoticeEvent() {
    /**
     * set、unset
     * 事件子类型, 分别表示设置和取消管理
     */
    @SerializedName( "sub_type")
    var subType = ""

    /**
     * 群号
     */
    @SerializedName( "group_id")
    var groupId = 0L

    // 管理员QQ号 userId 已在 NoticeEvent 中实现
}
